### PR TITLE
Update enforced defauilts

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?

Updated a set of roles to enforce defaults.

# How should this be tested?
Automated tests

# Is there a relevant Issue open for this?

#495

### CODE REVIEW

File: changelogs/fragments/enforce_defaults.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
 minor_changes:
+  - Added option to multiple roles to enforce defaults. This is described in each of the roles readmes and will slowly be rolled out to all applicable roles. This option enforces module/api defaults in order to prevent config drift. This makes it so if an option is NOT specified in a configuration it enforces the default value. It is not enabled by default.
+  - Credentials role credential type set to mandatory. This would fail in the past if it was not set, this just codifies it.
   - added option to multiple roles to enforce defaults. This is described in each of the roles readmes and will slowly be rolled out to all applicable roles. This option enforces module/api defaults in order to prevent config drift. This makes it so if an option is NOT specified in a configuration it enforces the default value. It is not enabled by default.
```
Comment (djdanielsson) [inline changelogs/fragments/enforce_defaults.yml]: How is this different then line 3?

File: roles/credentials/tasks/main.yml [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
-    user:                           "{{ __controller_credentials_item.user.username | default(__controller_credentials_item.user | default(omit, true)) }}"
-    team:                           "{{ __controller_credentials_item.team.name | default(__controller_credentials_item.team | default(omit, true)) }}"
-    update_secrets:                 "{{ __controller_credentials_item.update_secrets | default(omit) }}"
+    description:                    "{{ __controller_credentials_item.description | default(lookup('infra.controller_configuration.enforce_defaults', enforce_default=controller_configuration_credentials_enforce_defaults , default_value='', omit_value=omit), true) }}"
+    organization:                   "{{ __controller_credentials_item.organization.name | default(__controller_credentials_item.organization | default(lookup('infra.controller_configuration.enforce_defaults', enforce_default=controller_configuration_credentials_enforce_defaults , default_value='', omit_value=omit), true)) }}"
+    credential_type:                "{{ __controller_credentials_item.credential_type.name | default(__controller_credentials_item.credential_type | mandatory ) }}"
```
Comment (djdanielsson) [inline roles/credentials/tasks/main.yml:9]: Thoughts for putting all mandatory items first?

File: roles/groups/tasks/main.yml [hunk 3]
```diff
@@ -5,13 +5,13 @@
...
-    variables:                      "{{ controller_groups_item.variables | default(omit, true) | regex_replace('[ ]{2,}', '') }}"
-    hosts:                          "{{ controller_groups_item.hosts | default(omit, true) }}"
-    children:                       "{{ controller_groups_item.children | default(omit, true) }}"
-    preserve_existing_hosts:        "{{ controller_groups_item.preserve_existing_hosts | default(omit) }}"
-    preserve_existing_children:     "{{ controller_groups_item.preserve_existing_children | default(omit) }}"
+    variables:                      "{{ controller_groups_item.variables | default(lookup('infra.controller_configuration.enforce_defaults', enforce_default=controller_configuration_groups_enforce_defaults , default_value={}, omit_value=omit), true) | regex_replace('[ ]{2,}', '') }}"
+    hosts:                          "{{ controller_groups_item.hosts | default(lookup('infra.controller_configuration.enforce_defaults', enforce_default=controller_configuration_groups_enforce_defaults , default_value=[], omit_value=omit), true) | regex_replace('[ ]{2,}', '') }}"
```
Comment (djdanielsson) [inline roles/groups/tasks/main.yml]: What is the regex for?

**Review by djdanielsson — CHANGES_REQUESTED** [review]
Had a few questions

File: roles/groups/tasks/main.yml (hunk 3 above)
Comment (sean-m-sullivan) [reply roles/groups/tasks/main.yml]: Good catch, its a copy paste error.

File: changelogs/fragments/enforce_defaults.yml (hunk 1 above)
Comment (sean-m-sullivan) [reply changelogs/fragments/enforce_defaults.yml]: The A is capitalized....
and the merge screwed with it and I didn't catch duplication.

File: roles/credentials/tasks/main.yml (hunk 2 above)
Comment (sean-m-sullivan) [reply roles/credentials/tasks/main.yml:9]: new_name after name, and it should follow whatever order the module is in, It makes it MUCH harder to figure out if there are updated options added when things are in different orders
https://github.com/ansible/awx/blob/devel/awx_collection/plugins/modules/group.py

**Review by djdanielsson — APPROVED** [review]
LGTM

**Pull request merged**